### DISCUSSION TITLE
Is there any Task status listener for Completion and Failure of system task?

### DISCUSSION THREAD

**Discussion by User-Nikita** [Netflix/conductor#2322] [Q&A]
**Is there any Task status listener in Conductor just like a workflowStatusListener?** 
If yes, please guide me how to use it.
Details about my question-
In my workflow definition, **all tasks are System tasks.** 
I am using Java program for handling the workflow.
I want to **give notification to Java program** about status of each System task in workflow.
How can I solve this problem?

Thank you in advance..

**Comment by aravindanr** [comment]
Hello, Conductor currently does not have task level notifications. However it is part of our [roadmap](https://github.com/Netflix/conductor/wiki/Roadmap). Since its not in the top of our priority list, its **unlikely** that it will added in 2021.

**Reply by User-Nikita** [reply]
Thank you for your response. 
Can you please guide me what changes or additions should I do in source code to give task level notification ?

**Reply by apanicker-nflx** [reply]
We suggest you to walk through the source code and isolate out the modules that are responsible for updating the state of a task within a given workflow. Some key classes where this is done are: `WorkflowExecutor`, `DeciderService`, `DefaultActionProcessor` etc. You would also need to consider delivery guarantees for these notifications and how they impact the workflow execution/task scheduling. 

Please open a discussion with a design proposal for this feature and we can work out the design. Thanks.
